src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -114,27 +118,31 @@ void CachingReaderWorker::run() {
     }
 }
 
-void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
+void CachingReaderWorker::cleanUpOldTrack() {
```

Comment:
`closeAudioSource()`? We are not dealing with a "track" here, just with its audio source.

I didn't check if it would make sense to introduce a corresponding `openAudioSource()` function.